### PR TITLE
Upgrade Rubocop to 0.93

### PR DESCRIPTION
**Pull request by teceler**
* Upgrade Rubocop to 0.93
* Upgrade rubocop-performance to 1.10.2 
* Upgrade rubocop-rails to 2.9.1
* Upgrade rubocop-rspec to 1.44.1 

New rubocop feature: cops introduced since the last major version (or this change's introduction) must be manually enabled.

**New (enabled) Layout cops:**
Layout/BeginEndAlignment
Layout/SpaceAroundMethodCallOperator

**New (enabled) Lint cops:**
Lint/BinaryOperatorWithIdenticalOperands
Lint/ConstantDefinitionInBlock
Lint/DeprecatedOpenSSLConstant
Lint/DuplicateElsifCondition
Lint/DuplicateRequire
Lint/DuplicateRescueException
Lint/EmptyConditionalBody
Lint/EmptyFile
Lint/FloatComparison
Lint/HashCompareByIdentity
Lint/IdentityComparison
Lint/MissingSuper
Lint/MixedRegexpCaptureTypes
Lint/OutOfRangeRegexpRef
Lint/RaiseException
Lint/RedundantSafeNavigation
Lint/SelfAssignment
Lint/StructNewOverride
Lint/TopLevelReturnWithArgument
Lint/TrailingCommaInAttributeDeclaration
Lint/UnreachableLoop
Lint/UselessMethodDefinition
Lint/UselessTimes

**New (enabled) Style cops:**
Style/AccessorGrouping
Style/BisectedAttrAccessor
Style/CaseLikeIf
Style/ClassEqualityComparison
Style/CombinableLoops
Style/HashLikeCase
Style/RedundantAssignment
Style/RedundantFetchBlock
Style/RedundantFileExtensionInRequire
Style/RedundantRegexpCharacterClass
Style/RedundantRegexpEscape
Style/RedundantSelfAssignment
Style/SingleArgumentDig
Style/SoleNestedConditional

**New (enabled) Performance cops:**
Performance/AncestorsInclude
Performance/BigDecimalWithNumericArgument
Performance/BindCall
Performance/BlockGivenWithExplicitBlock
Performance/ConstantRegexp
Performance/DeletePrefix
Performance/DeleteSuffix
Performance/MethodObjectAsBlock
Performance/RedundantEqualityComparisonBlock
Performance/RedundantSortBlock
Performance/RedundantSplitRegexpArgument
Performance/RedundantStringChars
Performance/ReverseFirst
Performance/SortReverse
Performance/Squeeze
Performance/StringInclude
Performance/Sum

**New (enabled) Rails cops:**
Rails/ActiveRecordCallbacksOrder
Rails/AfterCommitOverride
Rails/AttributeDefaultBlockValue
Rails/FindById
Rails/IndexBy
Rails/IndexWith
Rails/Inquiry
Rails/MailerName
Rails/MatchRoute
Rails/NegateInclude
Rails/Pluck
Rails/SquishedSQLHeredocs
Rails/RenderInline
Rails/RenderPlainText
Rails/ShortI18n
Rails/WhereEquals
Rails/WhereNot

**New (enabled) RSpec cops:**
Capybara/VisibilityMatcher
RSpec/EmptyHook
RSpec/VariableName
RSpec/VariableDefinition
RSpec/RepeatedIncludeExample

### CODE REVIEW

File: app/controllers/tags_controller.rb [hunk 1]
```diff
@@ -12,7 +12,7 @@ def index
     @post_counts = @post_counts.group('post_tags.tag_id').count
     @view = params[:view]
     @page_title = @view.present? ? @view.titlecase.pluralize : 'Tags'
-    @tag_options = (Tag::TYPES - ['GalleryGroup']).sort.reverse.map{|t| [t.titlecase, t]}.to_h
+    @tag_options = (Tag::TYPES - ['GalleryGroup']).sort.reverse.index_by{|t| t.titlecase}
```
Comment (Throne3d) [inline app/controllers/tags_controller.rb:15]:
```suggestion
    @tag_options = (Tag::TYPES - ['GalleryGroup']).sort.reverse.index_by(&:titlecase)
```

File: .rubocop.yml [hunk 2]
```diff
@@ -190,18 +337,57 @@ Rails/HelperInstanceVariable:
...
+Rails/SaveBang:
+  Enabled: true
+
+Rails/ShortI18n:
+  Enabled: true
+
+Rails/SquishedSQLHeredocs:
+  Enabled: false
+
+Rails/UniqueValidationWithoutIndex :
```
Comment (Throne3d) [inline .rubocop.yml:379]: Weird space on this one (was this config auto-generated?)
```suggestion
Rails/UniqueValidationWithoutIndex:
```

Comment (teceler) [reply .rubocop.yml:379]: I was copy-pasting from what rubocop gave me a lot but I think what one is actually my fault xD

File: spec/models/tag_spec.rb [hunk 3]
```diff
@@ -76,7 +76,7 @@ def create_tags
       tags = create_tags
       fetched = GalleryGroup.where(id: tags.map(&:id)).select(:id).ordered_by_id.with_character_counts
       expect(fetched).to eq(tags)
-      expect(fetched.map{|x| x[:character_count] }).to eq([0, 1, 2])
+      expect(fetched.map{|x| x[:character_count] }).to eq([0, 1, 2]) # rubocop:disable Rails/Pluck
```
Comment (Throne3d) [inline spec/models/tag_spec.rb:79]: Could use
```suggestion
      expect(fetched.to_a.pluck(:character_count)).to eq([0, 1, 2])
```
I guess? But it doesn't seem like it's much clearer imo

**Review by Throne3d — APPROVED** [review]
LGTM!

**Pull request merged**